Validation/HcalDigis/src/HcalDigisValidation.cc

```diff
@@ -154,112 +194,100 @@ void HcalDigisValidation::booking(DQMStore::IBooker &ib, const std::string bsubd
     HistLim ietaLim(85, -42.5, 42.5);
     HistLim iphiLim(74, -0.5, 73.5);
 
-    if (bsubdet == "HE") {
+    if (bsubdet == "HB") {
+        Ndigis = HistLim( ((int)(nChannels_[1]/100) + 1)*100, 0., (float)((int)(nChannels_[1]/100) + 1)*100);
+    } else if (bsubdet == "HE") {
         sime = HistLim(200, 0., 1.0);
+        Ndigis = HistLim( ((int)(nChannels_[2]/100) + 1)*100, 0., (float)((int)(nChannels_[2]/100) + 1)*100);
     } else if (bsubdet == "HF") {
         sime = HistLim(100, 0., 100.);
         pedLim = HistLim(100, 0., 20.);
         pedWidthLim = HistLim(100, 0., 5.);
         frac = HistLim(400, -4.00, 4.00);
+        Ndigis = HistLim( ((int)(nChannels_[4]/100) + 1)*100, 0., (float)((int)(nChannels_[4]/100) + 1)*100);
     } else if (bsubdet == "HO") {
         sime = HistLim(200, 0., 1.0);
         gainLim = HistLim(160, 0., 1.6);
+        Ndigis = HistLim( ((int)(nChannels_[3]/100) + 1)*100, 0., (float)((int)(nChannels_[3]/100) + 1)*100);
     }
 
+    int isubdet=0;
+    if      (bsubdet=="HB") isubdet=1;
+    else if (bsubdet=="HE") isubdet=2;
+    else if (bsubdet=="HO") isubdet=3;
+    else if (bsubdet=="HF") isubdet=4;
+    else std::cout << "Warning: not HB/HE/HF/HO " << bsubdet << std::endl;
```

Comment:
Please don't use std::cout, use edm::LogWarning.
Even if it is as simple as "invalid parameter" or "product not found" (in the couts bellow), it can easily flood the production log.